src/libs/NextStepUtils.ts

```diff
@@ -78,7 +79,16 @@ function buildNextStep(report: OnyxEntry<Report>, predictedNextStatus: ValueOf<t
     const autoReportingFrequency = PolicyUtils.getCorrectedAutoReportingFrequency(policy);
     const submitToAccountID = PolicyUtils.getSubmitToAccountID(policy, ownerAccountID);
     const ownerDisplayName = ReportUtils.getDisplayNameForParticipant(ownerAccountID);
-    const managerDisplayName = ReportUtils.getDisplayNameForParticipant(submitToAccountID);
+    let nextApproverDisplayName = ReportUtils.getDisplayNameForParticipant(submitToAccountID);
+
+    const approvalChain = PolicyUtils.getApprovalChain(policy, ownerAccountID, report?.total ?? 0);
```

Comment:
I think it makes sense for this block to be a separate method  say `getNextApproverDisplayName`, wdyt?

Reply:
```
function getNextApproverDisplayName(policy, ownerAccountID, report, currentUserEmail, submitToAccountID) {
    const approvalChain = PolicyUtils.getApprovalChain(policy, ownerAccountID, report?.total ?? 0);
    
    if (approvalChain.length === 0) {
        return ReportUtils.getDisplayNameForParticipant(submitToAccountID);
    }

    const nextApproverEmail = approvalChain.length === 1 
        ? approvalChain[0] 
        : approvalChain[approvalChain.indexOf(currentUserEmail) + 1];

    if (!nextApproverEmail) {
        return ReportUtils.getDisplayNameForParticipant(submitToAccountID);
    }

    return PersonalDetailsUtils.getPersonalDetailByEmail(nextApproverEmail)?.displayName ?? nextApproverEmail;
}
```

Reply:
for readability that seems like a good suggestion. 

but this function will never be re-used. I'll defer to @Beamanator to decide this one for us.

Reply:
I'm slightly leaning toward agreeing with @mananjadhav  - I also doubt this function will be reused, BUT if `buildNextStep` follows the trajectory of our backend code that does similar calculation, it gunna get BIG so it would be great to make some functions that can keep that code smallish now, before it's too late and needs epic refactoring